pkg/utils/kubernetes/managedseed.go

```diff
@@ -44,3 +45,20 @@ func GetManagedSeed(ctx context.Context, seedManagementClient gardenseedmanageme
 	}
 	return &managedSeedList.Items[0], nil
 }
+
+// GetManagedSeedWithReader gets the ManagedSeed resource for the given shoot namespace and name,
+// by searching for all ManagedSeeds in the shoot namespace that have spec.shoot.name set to the shoot name.
+// If no such ManagedSeeds are found, nil is returned.
+func GetManagedSeedWithReader(ctx context.Context, r client.Reader, shootNamespace, shootName string) (*seedmanagementv1alpha1.ManagedSeed, error) {
```

Comment:
Can the old function (`GetManagedSeed`) be deleted?

Reply:
Nope, that one is still used by admission plugins.